internal/rest/leveldbreceipts.go

```diff
@@ -0,0 +1,373 @@
+// Copyright 2021 Kaleido
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/kaleido-io/ethconnect/internal/errors"
+	"github.com/kaleido-io/ethconnect/internal/kvstore"
+	"github.com/oklog/ulid/v2"
+	log "github.com/sirupsen/logrus"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type levelDBReceipts struct {
+	conf         *LevelDBReceiptStoreConf
+	store        kvstore.KVStore
+	idEntropy    *ulid.MonotonicEntropy
+	defaultLimit int
+}
+
+func newLevelDBReceipts(conf *LevelDBReceiptStoreConf) (*levelDBReceipts, error) {
+	store, err := kvstore.NewLDBKeyValueStore(conf.Path)
+	if err != nil {
+		return nil, errors.Errorf(errors.ReceiptStoreLevelDBConnect, err)
+	}
+	t := time.Unix(1000000, 0)
+	entropy := ulid.Monotonic(rand.New(rand.NewSource(t.UnixNano())), 0)
```

Comment:
Was reading this in the Godoc:
```
Please note that rand.Rand from the math package is not safe for concurrent use. Instantiate one per long living go-routine or use a sync.Pool if you want to avoid the potential contention of a locked rand.Source as its been frequently observed in the package level functions.
```

Reply:
This conversation was useful: https://github.com/cnabio/cnab-go/pull/207/files#r424489120
Sounds like the most straight forward safe implementation is a simple mutex around the generation (or probably more easily, the `AddReceipt function